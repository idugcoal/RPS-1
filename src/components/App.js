import React, { Component } from 'react';
import PlayerCounter from './PlayerCounter';
import ComputerCounter from './ComputerCounter';
import CurrentResult from './CurrentResult';
import Rock from './Rock';
import Paper from './Paper';
import Scissors from './Scissors';
import { simulateComputer, compare } from '../utils';
import { ROCK, PAPER, SCISSORS } from '../constants';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      playerWins: 0,
      computerWins: 0,
    };
    this.weaponClick = this.weaponClick.bind(this);
  }

  weaponClick(weapon) {
    const computerSelection = simulateComputer();
    // 1 means first argument wins, -1 means second argument wins, 0 is tie
    const winner = compare(weapon, computerSelection);
    this.setState(({ playerWins, computerWins }) => (
      {
        playerSelection: weapon,
        computerSelection,
        playerWins: winner === 1 ? playerWins + 1 : playerWins,
        computerWins: winner === -1 ? computerWins + 1 : computerWins,
      }
    ));
  }
  render() {
    const {
      playerWins,
      computerWins,
      playerSelection,
      computerSelection,
    } = this.state;

    return (
      <div>
        <PlayerCounter wins={playerWins} />
        <ComputerCounter wins={computerWins} />
        <CurrentResult
          playerSelection={playerSelection}
          computerSelection={computerSelection}
        />
        <div>
          <Rock onClick={() => this.weaponClick(ROCK)} />
          <Paper onClick={() => this.weaponClick(PAPER)} />
          <Scissors onClick={() => this.weaponClick(SCISSORS)} />
        </div>
      </div>
    );
  }
}
