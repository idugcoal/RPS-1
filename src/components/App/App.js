import React, { Component } from 'react';
import PlayerMode from '../PlayerMode/PlayerMode';
import SimulationMode from '../SimulationMode/SimulationMode';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      gameMode: 'player',
    };
    this.toggleMode = this.toggleMode.bind(this);
  }

  toggleMode() {
    const { gameMode } = this.state;
    this.setState({
      gameMode: gameMode === 'player' ? 'simulation' : 'player',
    });
  }

  render() {
    const { gameMode } = this.state;
    return (
      gameMode === 'player' ? <PlayerMode toggleMode={this.toggleMode} /> :
      <SimulationMode toggleMode={this.toggleMode} />
    );
  }
}
