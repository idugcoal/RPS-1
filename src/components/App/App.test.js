import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

test('Should render correctly initially', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});

test('Should render correctly based on state', () => {
  const wrapper = shallow(<App />);
  wrapper.setState({ gameMode: 'simulation' });
  expect(wrapper).toMatchSnapshot();
});
