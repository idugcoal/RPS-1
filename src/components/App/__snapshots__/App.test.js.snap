// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly based on state 1`] = `
<SimulationMode
  onToggle={[Function]}
/>
`;

exports[`Should render correctly initially 1`] = `
<PlayerMode
  onToggle={[Function]}
/>
`;
